### PR TITLE
IGNITE-19396 Move work with indexes from StorageUpdateHandler to a separate class

### PR DESCRIPTION
**Pull request by tkalkirill**
https://issues.apache.org/jira/browse/IGNITE-19396

### CODE REVIEW

File: modules/storage-api/src/main/java/org/apache/ignite/internal/storage/index/IndexStorage.java [hunk 1]
```diff
@@ -67,7 +67,7 @@ public interface IndexStorage {
     @Nullable RowId getNextRowIdToBuild() throws StorageException;
 
     /**
-     * Sets the row ID for which the index needs to be built, {@code null} means that the index is built.
+     * Sets the row ID for which the index needs to be build, {@code null} means that the index is build.
```
Comment (rpuch) [inline modules/storage-api/src/main/java/org/apache/ignite/internal/storage/index/IndexStorage.java:70]: Let's revert this change, because 'to be built' seems to be the correct form ('built' is the 3rd form for 'to build')

File: modules/table/src/main/java/org/apache/ignite/internal/table/distributed/StorageUpdateHandler.java [hunk 2]
```diff
@@ -195,11 +193,11 @@ public void handleUpdateAll(
     void executeBatchGc() {
         HybridTimestamp lwm = lowWatermark.getLowWatermark();
 
-        if (lwm == null || safeTimeTracker.current().compareTo(lwm) < 0) {
+        if (lwm == null || gcUpdateHandler.getSafeTimeTracker().current().compareTo(lwm) < 0) {
```
Comment (rpuch) [inline modules/table/src/main/java/org/apache/ignite/internal/table/distributed/StorageUpdateHandler.java:196]: How about extracting the check to a method on GcUpdateHandler? It could be called `safeTimeBelow(HybridTimestamp lwm)` or similarly

Comment (tkalkirill) [reply modules/table/src/main/java/org/apache/ignite/internal/table/distributed/StorageUpdateHandler.java:196]: If you insist, I can add, but at the moment it seems that for a single method it is a bit redundant.

Comment (rpuch) [reply modules/table/src/main/java/org/apache/ignite/internal/table/distributed/StorageUpdateHandler.java:196]: It's up to you

**Review by rpuch — APPROVED** [review]
